solver/llbsolver/solver.go

```diff
@@ -227,7 +228,11 @@ func (s *Solver) Solve(ctx context.Context, id string, sessionID string, req fro
 			inp.Refs = m
 		}
 
-		if err := inBuilderContext(ctx, j, e.Name(), "", func(ctx context.Context, _ session.Group) error {
+		var pg *pb.ProgressGroup
+		if res.Ref != nil {
+			pg = res.Ref.ProgressGroup()
```

Comment:
Not sure why you need to track progressgroup through the result when something like this is not needed for the name? To me, it should just be like a "secondary name" so implementation almost identical on daemon side.

Reply:
This was the only way I could find to plumb the progress group metadata through to the `inBuilderContext` call below, which as mentioned in the other comment is needed so that progress groups show up correctly when an export forces an unlazy. I'm very open to suggestions on different ways of plumbing it.

> To me, it should just be like a "secondary name" so implementation almost identical on daemon side.

I'm not following, can you explain in more detail.